addons/mrp_subcontracting/models/mrp_production.py

```diff
@@ -1,20 +1,64 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import fields, models
+from collections import defaultdict
+from odoo import fields, models, _, api
+from odoo.exceptions import UserError
 from odoo.tools.float_utils import float_compare, float_is_zero
 
 
 class MrpProduction(models.Model):
     _inherit = 'mrp.production'
 
-    def write(self, vals):
-        res = super().write(vals)
-        if self.env.context.get('subcontract_move_id'):
-            for production in self:
-                if production.state not in ('cancel', 'done') and \
-                        ('lot_producing_id' in vals or 'qty_producing' in vals):
-                    production._update_finished_move()
+    move_line_raw_ids = fields.One2many(
+        'stock.move.line', string="Detail Component", readonly=False,
+        inverse='_inverse_move_line_raw_ids', compute='_compute_move_line_raw_ids'
+    )
+
+    @api.depends('move_raw_ids.move_line_ids')
+    def _compute_move_line_raw_ids(self):
+        for production in self:
+            production.move_line_raw_ids = production.move_raw_ids.move_line_ids
+
+    def _inverse_move_line_raw_ids(self):
+        for production in self:
+            line_by_product = defaultdict(lambda: self.env['stock.move.line'])
+            for line in production.move_line_raw_ids:
+                line_by_product[line.product_id] |= line
+            for move in production.move_raw_ids:
+                move.move_line_ids = line_by_product.pop(move.product_id, self.env['stock.move.line'])
+            for product_id, lines in line_by_product.items():
+                qty = sum(line.product_uom_id._compute_quantity(line.qty_done, product_id.uom_id) for line in lines)
+                move = production._get_move_raw_values(product_id, qty, product_id.uom_id)
+                move['additional'] = True
+                production.move_raw_ids = [(0, 0, move)]
```

Comment:
move['move_line_ids'] = [(4, ml.id) for ml in lines]
and remove
production.move_raw_ids.filtered(lambda m: m.product_id == product_id)[:1].move_line_ids = lines
?

Reply:
Obviously, It doesn't work. I don't know the real reason behind that :/